src/org/labkey/targetedms/parser/skyd/ChromGroupHeaderInfo.java

```diff
@@ -45,52 +45,56 @@
+        // absence of accessors and reading from inputStream is left to correctly advance the size of the fields
         if (cacheFormatVersion.compareTo(CacheFormatVersion.Five) < 0) {
             precursor = Float.intBitsToFloat(dataInputStream.readInt());
             fileIndex = checkUShort(dataInputStream.readInt());
             numTransitions = checkUShort(dataInputStream.readInt());
             startTransitionIndex = dataInputStream.readInt();
-            numPeaks = checkByte(dataInputStream.readInt());
-            startPeakIndex = dataInputStream.readInt();
-            int maxPeakIndexInt = dataInputStream.readInt();
-            maxPeakIndex = maxPeakIndexInt == -1 ? (byte) 0xff : checkByte(maxPeakIndexInt);
+            /*numPeaks =*/ checkByte(dataInputStream.readInt());
+            /*startPeakIndex =*/ dataInputStream.readInt();
+            /*int maxPeakIndexInt = */dataInputStream.readInt();
+            /*maxPeakIndex = maxPeakIndexInt == -1 ? (byte) 0xff : checkByte(maxPeakIndexInt);*/
             numPoints = dataInputStream.readInt();
             compressedSize = dataInputStream.readInt();
             dataInputStream.readInt(); // ignore these four bytes
             locationPoints = dataInputStream.readLong();
         } else {
             textIdIndex = dataInputStream.readInt();
             startTransitionIndex = dataInputStream.readInt();
-            startPeakIndex = dataInputStream.readInt();
-            startScoreIndex = dataInputStream.readInt();
+            /*startPeakIndex =*/ dataInputStream.readInt();
+            /*startScoreIndex =*/ dataInputStream.readInt();
             numPoints = dataInputStream.readInt();
             compressedSize = dataInputStream.readInt();
             flagBits = dataInputStream.readShort();
             fileIndex = dataInputStream.readShort();
             textIdLen = dataInputStream.readShort();
             numTransitions = dataInputStream.readShort();
-            numPeaks = dataInputStream.readByte();
-            maxPeakIndex = dataInputStream.readByte();
-            isProcessedScans = dataInputStream.readByte();
-            align1 = dataInputStream.readByte();
-            statusId = dataInputStream.readShort();
-            statusRank = dataInputStream.readShort();
+            /*numPeaks =*/ dataInputStream.readByte();
+            /*maxPeakIndex =*/ dataInputStream.readByte();
+            /*isProcessedScans =*/ dataInputStream.readByte();
+            /*align1 =*/ dataInputStream.readByte();
+            /*statusId =*/ dataInputStream.readShort();
+            /*statusRank =*/ dataInputStream.readShort();
             precursor = dataInputStream.readDouble();
             locationPoints = dataInputStream.readLong();
         }
         if (cacheFormatVersion.compareTo(CacheFormatVersion.Eleven) < 0) {
             uncompressedSize = -1;
+            startTime = Float.MIN_VALUE;
```

Comment:
Float.MIN_VALUE is a tiny but positive number. I suggest using Float.NEGATIVE_INFINITY instead, and assigning it to a static final variable called NULL_PLACEHOLDER or something like that with a comment that it's used to reduce the memory footprint